server/src/utils/userUtils.js

```diff
@@ -12,7 +13,15 @@ const createUser = (
 
 const removeUserFromRoom = (user: UserT, rooms: RoomSetT): void => {
   const room = getUserRoom(user, rooms);
-  rooms[room.id].users = room.users.filter(name => name != user.username);
+  const roomv2: RoomT = 
```

Comment:
I think it's easier to understand if we remove the user and then check if it was the owner, that way we will only need one aux room. That way we can easily see that removeUserFromRoom function removes the user and then, if it was the owner, transfers it.